lib/net.js

```diff
@@ -2019,6 +2020,10 @@ Server.prototype.listen = function(...args) {
     toNumber(args.length > 2 && args[2]);  // (port, host, backlog)
 
   options = options._handle || options.handle || options;
+  if (typeof options.host === 'string' && RegExpPrototypeExec(/\[.*\]/, options.host) !== null) {
```

Comment:
Why not doing the opposite? I mean testing for allowed characters `/^[a-zA-Z0-9.:-]+$/`. It does not really matter but `/\[.*\]/` is also vulnerable to ReDoS.

Reply:
Good point. I was too focused on explaining the special case from the issue.